heat-core-utils/src/main/java/com/hotels/heat/core/checks/BasicChecks.java

```diff
@@ -120,11 +119,12 @@ private boolean checkResponseCode(Map testCaseParams) {
         }
         if (testCaseParams.containsKey(EXPECTS_JSON_ELEMENT)) {
             Map<String, String> expectedParams = (Map<String, String>) testCaseParams.get(EXPECTS_JSON_ELEMENT);
-            // The attribute "responseCode" in the json file is optional. If it is missing, the default value is 200
-            int expectedRespCode = expectedParams.get(RESPONSE_CODE_JSON_ELEMENT) != null ? Integer.parseInt(expectedParams.get(RESPONSE_CODE_JSON_ELEMENT)) : HttpStatus.SC_OK;
-            int currentStatusCode = ((Response) responses).getStatusCode();
-            logUtils.debug("check response code: current '{}' / expected '{}'", currentStatusCode, expectedRespCode);
-            isCheckOk &= assertionHandler.assertion(isBlocking, "assertEquals", logUtils.getTestCaseDetails() + "{checkResponseCode} ", currentStatusCode, expectedRespCode);
+            String expectedRespCode = expectedParams.get(RESPONSE_CODE_JSON_ELEMENT);
```

Comment:
Why code is now string?

Reply:
We chose to handle the response code as a string because it avoids the eventuality of a NumberFormatException when parsing and the code seems to be more straightforward.